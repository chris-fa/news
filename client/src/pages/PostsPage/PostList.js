import React from 'react';

const PostList = ({ posts }) => (
  <div>
    { posts.map(post =>
      <div key={ post.id } className="box">
        <h4 className="title is-4">{ post.title }</h4>
        <h6 className="subtitle is-6">{ `By ${post.author} at
          ${post.updatedAt.toLocaleDateString('en-US')}` }</h6>

        <figure className="image is-4by3">
          <img src={ post.image } alt={ post.title }/>
        </figure>

        <p>{ post.body }</p>
      </div>
    )}
  </div>
);

export default PostList;
