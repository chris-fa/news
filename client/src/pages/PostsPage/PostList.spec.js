import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PostList from './PostList';

describe('<PostList />', () => {
  const defaultProps = {
    posts: [{
      id: 1,
      title: 'foo',
      body: 'bar',
      author: 'John Doe',
      updatedAt: new Date(2018, 3, 18),
    }, {
      id: 2,
      title: 'foo 2',
      body: 'bar 2',
      author: 'Jane Doe',
      updatedAt: new Date(2018, 3, 19),
    }],
  };

  it('renders posts', () => {
    const props = defaultProps;

    const wrapper = shallow(<PostList {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.box')).toHaveLength(2);
  });
});
