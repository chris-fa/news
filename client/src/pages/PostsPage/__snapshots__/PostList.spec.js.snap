// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> renders posts 1`] = `
<div>
  <div
    className="box"
    key="1"
  >
    <h4
      className="title is-4"
    >
      foo
    </h4>
    <h6
      className="subtitle is-6"
    >
      By John Doe at
                4/18/2018
    </h6>
    <figure
      className="image is-4by3"
    >
      <img
        alt="foo"
      />
    </figure>
    <p>
      bar
    </p>
  </div>
  <div
    className="box"
    key="2"
  >
    <h4
      className="title is-4"
    >
      foo 2
    </h4>
    <h6
      className="subtitle is-6"
    >
      By Jane Doe at
                4/19/2018
    </h6>
    <figure
      className="image is-4by3"
    >
      <img
        alt="foo 2"
      />
    </figure>
    <p>
      bar 2
    </p>
  </div>
</div>
`;
